### PR TITLE
Feat/#252 - kumactl get for single entities

### LINKED ISSUE #252: Introduced `describe` as a new CLI command for our entities
For example:

```sh
$ kumactl describe dp-echo-1 --mesh default
type: Dataplane
mesh: default
name: backend-1
networking:
  inbound:
  - interface: 127.0.0.1:8000:80
    tags:
      service: backend
  outbound:
  - interface: :10000
    service: redis
```

### PR DESCRIPTION
**Pull request by tharun208**
### Summary
kumactl get command for single entities
### Full changelog

* Implemented command to get information about a single entity of Kuma for the following
		
	 * [x]  Fault Injection
	 * [x] Dataplane
	 * [x] Mesh
	 * [x] Health Check
	 * [x] proxyTemplate
	 * [x] trafficLog
	 * [x] trafficPermission
	 * [x] trafficRoute
	 * [x] trafficTrace
* Added tests
* Updated tests for `kumactl apply`

### Issues resolved

Fix #252

### CODE REVIEW

**Comment by tharun208** [comment]
And we have commands like get meshes, get data planes, can I just create a `get mesh mesh-name` for now and later, can we think of a better approach ?.

Is this fine because if I start with the `get` root command to handle args, it will break our existing implementation of underlying sub-commands.

**Comment by jakubdyszkiewicz** [comment]
Yes, I think the right approach for now is to create a separate sub commands `get dataplane`, `get mesh` etc.

File: app/kumactl/cmd/get/get_mesh.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+			}
+		},
+	}
+	return cmd
+}
+
+func printMesh(mesh *mesh.MeshResource, out io.Writer) error {
+	data := printers.Table{
+		Headers: []string{"MESH", "NAME"},
+		NextRow: func() []string {
```
Comment (lobkovilya) [inline app/kumactl/cmd/get/get_mesh.go:57]: I guess `NextRow` should return `nil`, after the last element. Please take a look here: https://github.com/Kong/kuma/blob/master/app/kumactl/cmd/get/get_dataplanes.go#L48

Comment (tharun208) [reply app/kumactl/cmd/get/get_mesh.go]: Thanks @lobkovilya fixed it

**Comment by tharun208** [comment]
@jakubdyszkiewicz I have added the implementation for `get mesh` and `get dataplane`

**Comment by tharun208** [comment]
I am not finding any possible abstraction between these two. I thought of abstracting out `printTable` but it should be specific to each resource type.

**Comment by jakubdyszkiewicz** [comment]
Hey, yes please - reuse the same function for printing table. As you can see there is a difference already, `get dataplane` does not return tags where as `get dataplanes` returns tags.

For tests I think we can reuse YAML file for input and just have a one parametrized test https://github.com/Kong/kuma/commit/b4e56abe2bb81ec65dd9136f8a6fee8548175a90

**Comment by tharun208** [comment]
> Hey, yes please - reuse the same function for printing table. As you can see there is a difference already, `get dataplane` does not return tags where as `get dataplanes` returns tags.
> 
> For tests I think we can reuse YAML file for input and just have a one parametrized test [b4e56ab](https://github.com/Kong/kuma/commit/b4e56abe2bb81ec65dd9136f8a6fee8548175a90)

If I need to reuse the existing methods means, I need to wrap my `resource` type into `resourceTypeList` e.g: `meshResource` into `meshResourceList` to make it work. Is this fine?

**Comment by tharun208** [comment]
> Hey, yes please - reuse the same function for printing table. As you can see there is a difference already, `get dataplane` does not return tags where as `get dataplanes` returns tags.
> 
> For tests I think we can reuse YAML file for input and just have a one parametrized test [b4e56ab](https://github.com/Kong/kuma/commit/b4e56abe2bb81ec65dd9136f8a6fee8548175a90)

only `-oyaml` file test is enough for this, no need of table and json ?

File: app/kumactl/cmd/get/get_fault_injections.go [hunk 2]
```diff
@@ -45,17 +45,17 @@ func newGetFaultInjectionsCmd(pctx *getContext) *cobra.Command {
 	return cmd
 }
 
-func printFaultInjection(faultInjections *mesh.FaultInjectionResourceList, out io.Writer) error {
+func printFaultInjection(faultInjections []*mesh.FaultInjectionResource, out io.Writer) error {
```
Comment (jakubdyszkiewicz) [inline app/kumactl/cmd/get/get_fault_injections.go:48]: I missed this when this func was added. Can we rename it to `printFaultInjections`?

File: app/kumactl/cmd/get/get_fault_injections.go [hunk 3]
```diff
@@ -45,17 +45,17 @@ func newGetFaultInjectionsCmd(pctx *getContext) *cobra.Command {
...
 		NextRow: func() func() []string {
 			i := 0
 			return func() []string {
 				defer func() { i++ }()
-				if len(faultInjections.Items) <= i {
+				if len(faultInjections) <= i {
 					return nil
 				}
-				faultInjections := faultInjections.Items[i]
+				faultInjections := faultInjections[i]
```
Comment (jakubdyszkiewicz) [inline app/kumactl/cmd/get/get_fault_injections.go:58]: `faultInjection := faultInjections[i]`

File: app/kumactl/cmd/get/get_proxytemplates.go [hunk 4]
```diff
@@ -45,17 +45,17 @@ func newGetProxyTemplatesCmd(pctx *getContext) *cobra.Command {
 	return cmd
 }
 
-func PrintProxyTemplates(proxyTemplates *mesh_core.ProxyTemplateResourceList, out io.Writer) error {
+func PrintProxyTemplates(proxyTemplates []*mesh_core.ProxyTemplateResource, out io.Writer) error {
```
Comment (jakubdyszkiewicz) [inline app/kumactl/cmd/get/get_proxytemplates.go:48]: can we make it private? `PrintProxyTemplates` -> `printProxyTemplates`

File: app/kumactl/cmd/get/get_traffic_logs.go [hunk 5]
```diff
@@ -45,17 +45,17 @@ func newGetTrafficLogsCmd(pctx *getContext) *cobra.Command {
 	return cmd
 }
 
-func printTrafficLog(trafficLogging *mesh.TrafficLogResourceList, out io.Writer) error {
+func printTrafficLog(trafficLogging []*mesh.TrafficLogResource, out io.Writer) error {
```
Comment (jakubdyszkiewicz) [inline app/kumactl/cmd/get/get_traffic_logs.go:48]: `printTrafficLog` -> `printTrafficLogs`

File: app/kumactl/cmd/get/testdata/get-mesh.golden.yaml [hunk 6]
```diff
@@ -0,0 +1,26 @@
+---
```
Comment (jakubdyszkiewicz) [inline app/kumactl/cmd/get/testdata/get-mesh.golden.yaml:1]: Can we get rid of this separator in every get-X.golden.yaml?

Comment (tharun208) [reply app/kumactl/cmd/get/testdata/get-mesh.golden.yaml:1]: was using a formatter will remove it

File: app/kumactl/cmd/get/testdata/get-traffic-route.golden.yaml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+mesh: default
```
Comment (jakubdyszkiewicz) [inline app/kumactl/cmd/get/testdata/get-traffic-route.golden.yaml:1]: Can we have a little bit more complete example of traffic route?

File: app/kumactl/cmd/get/testdata/get-proxytemplate.golden.yaml [hunk 8]
```diff
@@ -0,0 +1,3 @@
+mesh: default
+name: proxytemplate-1
+type: ProxyTemplate
```
Comment (jakubdyszkiewicz) [inline app/kumactl/cmd/get/testdata/get-proxytemplate.golden.yaml:3]: Can we have a little bit more complete example of proxy template?

File: app/kumactl/cmd/get/testdata/get-healthcheck.golden.yaml [hunk 9]
```diff
@@ -0,0 +1,3 @@
+mesh: default
+name: healthcheck-1
+type: HealthCheck
```
Comment (jakubdyszkiewicz) [inline app/kumactl/cmd/get/testdata/get-healthcheck.golden.yaml:3]: Can we have a little bit more complete example of health check?

File: app/kumactl/cmd/get/testdata/get-fault-injection.golden.yaml [hunk 10]
```diff
@@ -0,0 +1,12 @@
+---
```
Comment (jakubdyszkiewicz) [inline app/kumactl/cmd/get/testdata/get-fault-injection.golden.yaml:1]: I still see some separators. Can you please check all the YAML files that you add in this PR?

File: app/kumactl/cmd/get/get_dataplane.go [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+	"github.com/Kong/kuma/app/kumactl/pkg/output"
+	"github.com/Kong/kuma/app/kumactl/pkg/output/printers"
+	rest_types "github.com/Kong/kuma/pkg/core/resources/model/rest"
+	"github.com/Kong/kuma/pkg/core/resources/store"
+)
+
+func newGetDataplaneCmd(pctx *getContext) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "dataplane NAME",
+		Short: "Show Single Dataplane Resource",
```
Comment (jakubdyszkiewicz) [inline app/kumactl/cmd/get/get_dataplane.go:21]: how about `Show a single Dataplane resource`. No need for uppercase Single and Resource. Can you make this change in every command? Thank you.

**Comment by tharun208** [comment]
> It would help a lot with the review if the commits were more meaningful instead of same name for every one of them. Thanks!

sure noted it

**Comment by jakubdyszkiewicz** [comment]
Thank you for working on this!

**Pull request merged**